services/credits-service/src/services/billing_service.py

```
@@ -0,0 +1,157 @@
+        await self._ensure_system_account()
+
+        account_id = self.client.user_id_to_account_id(user_id)
+        amount_cents = int(amount * CURRENCY_PRECISION)
+
+        # Check current balance
+        current_balance_cents = await self.client.get_account_balance(account_id)
+        if current_balance_cents < amount_cents:
+            current_balance = Decimal(current_balance_cents) / CURRENCY_PRECISION
+            raise InsufficientBalanceException(
+                f"Insufficient balance. Current: ${current_balance}, Required: ${amount}"
+            )
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)